src/main/java/io/github/brassmc/brassloader/gui/ModsListScreen.java

```diff
@@ -184,35 +185,40 @@ public void render(@NotNull PoseStack poseStack, int mouseX, int mouseY, float p
-
-            poseStack.pushPose();
-            poseStack.scale(1.5f, 1.5f, 1.5f);
-            poseStack.translate((getBoxX() + getBoxWidth() / 2f) / 1.5f, getBoxY() + BOX_PADDING, 0f);
-            GuiComponent.drawCenteredString(poseStack, this.font, mod.name(), 0, 0, 0xFFFFFF);
-            poseStack.popPose();
-
-            GuiComponent.drawCenteredString(poseStack, this.font, mod.modid(), getBoxX() + getBoxWidth() / 2, getBoxY() + 35, 0xBABABA);
-            GuiComponent.drawCenteredString(poseStack, this.font, mod.version(), getBoxX() + getBoxWidth() / 2, getBoxY() + 45, 0x929292);
-
-            List<FormattedCharSequence> description = Language.getInstance().getVisualOrder(this.font.getSplitter().splitLines(mod.description(), getBoxWidth() - 20, Style.EMPTY));
-            int descriptionHeight = description.size() * this.font.lineHeight;
-            for (FormattedCharSequence line : description) {
-                int lineY = getBoxY() + BOX_PADDING + 45 + 10 + (this.font.lineHeight * description.indexOf(line));
-                if(lineY > getBoxY() && lineY < getBoxY() + getBoxHeight() - this.font.lineHeight) {
-                    GuiComponent.drawString(poseStack, this.font, line, getBoxX() + BOX_PADDING, lineY, 0xFFFFFF);
+            InformationPanel.Renderer renderer = InformationPanel.PANELS.get(mod.modid());
+
+            if (renderer != null) {
+                renderer.render(this, poseStack, mouseX, mouseY, partialTicks);
+            } else {
+                renderScrollbar(tesselator, buffer, getBoxX1() - 6, 200, 50, getBoxX(), getBoxY(), getBoxX1(), getBoxY1());
+
+                poseStack.pushPose();
+                poseStack.scale(1.5f, 1.5f, 1.5f);
+                poseStack.translate((getBoxX() + getBoxWidth() / 2f) / 1.5f, getBoxY() + BOX_PADDING, 0f);
+                GuiComponent.drawCenteredString(poseStack, this.font, mod.name(), 0, 0, 0xFFFFFF);
+                poseStack.popPose();
+
+                GuiComponent.drawCenteredString(poseStack, this.font, mod.modid(), getBoxX() + getBoxWidth() / 2, getBoxY() + 35, 0xBABABA);
+                GuiComponent.drawCenteredString(poseStack, this.font, mod.version(), getBoxX() + getBoxWidth() / 2, getBoxY() + 45, 0x929292);
+
+                List<FormattedCharSequence> description = Language.getInstance().getVisualOrder(this.font.getSplitter().splitLines(mod.description(), getBoxWidth() - 20, Style.EMPTY));
+                int descriptionHeight = description.size() * this.font.lineHeight;
+                for (FormattedCharSequence line : description) {
+                    int lineY = getBoxY() + BOX_PADDING + 45 + 10 + (this.font.lineHeight * description.indexOf(line));
+                    if (lineY > getBoxY() && lineY < getBoxY() + getBoxHeight() - this.font.lineHeight) {
+                        GuiComponent.drawString(poseStack, this.font, line, getBoxX() + BOX_PADDING, lineY, 0xFFFFFF);
+                    }
                 }
```

Comment:
> Use a defaulted map with that code as the default Renderer.

Where's a nice and pretty place to register the renderer?

Reply:
> Use a defaulted map with that code as the default Renderer.

I kinda can't or shouldn't, because the default code uses a ModContainer for a more abstract representation of a mod, whereas in a custom renderer things like the name, modid, version, etc would already be known by the creator. I could add a ModContainer parameter to the render method but that seems a bit redundant.

Reply:
> > Use a defaulted map with that code as the default Renderer.
> 
> I kinda can't or shouldn't, because the default code uses a ModContainer for a more abstract representation of a mod, whereas in a custom renderer things like the name, modid, version, etc would already be known by the creator. I could add a ModContainer parameter to the render method but that seems a bit redundant.

nvm, I have an idea